nixos/modules/services/web-apps/homebox.nix

```diff
@@ -10,19 +10,45 @@ let
     mkEnableOption
     mkPackageOption
     mkDefault
+    mkOption
     types
     mkIf
     ;
 in
 {
+  imports = [
+    (lib.mkRenamedOptionModule
+      [ "services" "homebox" "settings" ]
+      [ "services" "homebox" "environment" ]
+    )
+  ];
   options.services.homebox = {
     enable = mkEnableOption "homebox";
     package = mkPackageOption pkgs "homebox" { };
-    settings = lib.mkOption {
-      type = types.attrsOf types.str;
+    user = mkOption {
+      type = types.str;
+      default = "homebox";
+      description = "User account under which Homebox runs.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = "homebox";
+      description = "Group under which Homebox runs.";
+    };
+    connectionString = lib.mkOption {
+      type = types.str;
+      default = "file:///var/lib/homebox";
+      description = "Path to the Homebox data directory.";
+    };
```

Comment:
Please use a [freeform submodule](https://nixos.org/manual/nixos/stable/#sec-freeform-modules) option (i.e: `settings.HBOX_STORAGE_CONN_STRING` itself) for this option.

Reply:
Or do like the other defaulted `settings` values, set it with `mkDefault` and document it.